lib/Conf_t.mli

```diff
@@ -0,0 +1,138 @@
+(**************************************************************************)
```

Comment:
This module doesn't need a mli. It is 90% type definition, so it is 90% repeated code. The function in this modules are either trivial (eg. `to_string`) or shouldn't be there (eg. `warn_deprecated`).

Reply:
I agree that `warn_deprecated` is out of place here, but `Conf_decl` needs it. I will try to move it anyway.

Reply:
I think the diff would be smaller if we kept `Config_option` instead of renaming it into `Conf_decl` as it looks to me most of its content comes from `Config_option`.

Reply:
Yes, its the same module, thats why I put the renaming in a separate commit. 
I think the renaming is a good idea because it is a whole lot clearer. I had a lot of trouble understanding this part of the codebase because of this naming choice. If you dont think its a good idea to rename at all, we could remove the commit renaming it, or if the diff concerns you more, I could extract that commit to a separate pull request.

Reply:
For me `Config_option` made sense as the `'a Config_option.t` refers to an individual option and not the the configuration itself, and it obviously can't name it `Option`, I think we lose a bit of this information by using `Conf_decl`.

But I don't mind that much actually, I'm more concerned with the size of the diff. By renaming the file in a different PR, this PR will get much clearer.